src/traversals/bfs.jl

```diff
@@ -13,21 +13,19 @@ type BreadthFirst <: SimpleGraphVisitAlgorithm
 end
 
 function breadth_first_visit_impl!(
-    graph::SimpleGraph,   # the graph
-    queue::Vector{Int},                  # an (initialized) queue that stores the active vertices
-    colormap::Vector{Int},          # an (initialized) color-map to indicate status of vertices
-    visitor::SimpleGraphVisitor)  # the visitor
+    graph::SimpleGraph,                 # the graph
+    queue::Vector{Int},                 # an (initialized) queue that stores the active vertices
+    colormap,                           # an (initialized) color-map to indicate status of vertices (0=unseen, 1=seen, 2=closed)
```

Comment:
This violates the LG principal of having type annotations for all parameters. Can we do something else?